File: 283_move_zeroes/moveZeroes.go
package movezeroes

func moveZeroes(nums []int) {
	l := 0

	for r := 0; r < len(nums); r++ {
		if nums[l] == 0 {
			nums[l], nums[r] = nums[r], nums[l]
		}

		if nums[l] != 0 {
			l++
		}
	}
}

File: 283_move_zeroes/moveZeroes.cpp
#include <vector>
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int l = 0;

        for (int r = 0; r < nums.size(); r++) {
            if (nums[l] == 0) {
                int temp = nums[l];
                nums[l] = nums[r];
                nums[r] = temp;
            }
            if (nums[l] != 0)
                l++;
        }
    }
};